test/run-tests.sh

```diff
@@ -25,12 +25,20 @@ done
 # When updating subprojects, change the commit being checked out, and set the `shallow-since`
 # to the day before, to reduce the amount of refs being transferred and thus speed up CI.
 
+applyPatch () {
+	patch=../patches/${PWD##*/}.patch
+	if [ -f $patch ]; then
+		git apply $patch
```

Comment:
```suggestion
		git reset --hard
		git apply $patch
```

Reply:
Why? This is run right after doing a `checkout`.

Reply:
`git checkout` preserves modifications by default. I think this works because `git apply` applies against the index?

Reply:
(I think you'd want `git checkout -f` for that.)

Reply:
Sure. I got rid of the function abstraction too; there's so much repetition in the clone-pushd-fetch-checkout-make-popd commands for each project anyway.

Reply:
Ok, hear me out; what if:
```bash
test_downstream() { # owner/repo shallow-since commit make-target
	if [ ! -d ${1##*/} ]; then
		git clone https://github.com/$1.git --shallow-since=$2 --single-branch
	fi
	pushd ${1##*/}
	git checkout -f $3
	make clean
	make -j4 $4 RGBDS=../../
	popd
}
test_downstream pret/pokecrystal 2022-09-26 a9869f18962353b056559dc14dfc00fef0df5978 compare
test_downstream pret/pokered 2022-09-25 22859c4bb70dba17994c9b47b07f657ea082875d compare
test_downstream AntonioND/ucity 2022-04-20 d8878233da7a6569f09f87b144cb5bf140146a0f ''
```

Reply:
Sure. :P

Reply:
Wait, I forgot the patching part. Edited. Oops.

Reply:
(Doing a negation *then* `||` is intentional, as `[ ... ] && git ...` would return 1 if the test didn't pass, failing the script (`set -e`). I think. `set -e` [is fairly complicated](https://mywiki.wooledge.org/BashPitfalls#errexit), especially with functions.

Reply:
I had added that, but as the usual `if then fi`.

Reply:
Aww, where's the fun in that!? ;3

Reply:
(Otherwise you may as well do `[ -d ${1##*/} ] || git ...` and so on for other short conditionals.)

Reply:
File "code golf the shell scripts" as a separate issue. :P